Enabler: Move off the ubuntu-20.04 image in CI workflow

# Move CI workflow away from the Ubuntu 20.04 image in GitHub workflow.

Per email from GitHub, The `ubuntu-20.04` image will be retired on April 1.

We use it in [.github/workflows/workflow.yml](https://github.com/NCIOCPL/ncids/blob/1d0b3f50d57a80681b6911c3a948b0344eb79a5e/.github/workflows/workflow.yml#L18)

> The Ubuntu 20.04 runner image will be fully unsupported by April 1, 2025. To raise awareness of the upcoming removal, we will temporarily fail jobs using Ubuntu 20.04. Builds that are scheduled to run during the brownout periods will fail. The brownouts are scheduled for the following dates and times:
>
> - March 4 14:00 UTC – 22:00 UTC
> - March 11 13:00 UTC – 21:00 UTC
> - March 18 13:00 UTC – 21:00 UTC
> - March 25 13:00 UTC – 21:00 UTC
> 
> **What you need to do**
> 
> Jobs using the ubuntu-20.04 YAML workflow label should be updated to ubuntu-22.04, ubuntu-24.04 or ubuntu-latest. You can always get up-to-date information on our tools by reading about the software in the [runner images repository](https://github.com/actions/runner-images). Please contact GitHub [Support](https://support.github.com/request?tags=dotcom-contact-params) if you run into any problems or need help.

> **ESTIMATE** TBD

### Resources:


### Prerequisites


### Sub-Tasks


### Notes

There is some sort of problem between Puppeteer and Ubuntu `24.04`, however it works with `22.04`

```
> ncids-js-testing@3.3.0 test-pa11y /home/runner/work/ncids/ncids/testing/ncids-js-testing
> pa11y-ci

/home/runner/work/ncids/ncids/node_modules/.pnpm/puppeteer@9.1.1_encoding@0.1.13/node_modules/puppeteer/lib/cjs/puppeteer/node/BrowserRunner.js:194
            reject(new Error([
                   ^

Error: Failed to launch the browser process!
[0303/213045.901833:FATAL:zygote_host_impl_linux.cc(117)] No usable sandbox! Update your kernel or see https://chromium.googlesource.com/chromium/src/+/master/docs/linux/suid_sandbox_development.md for more information on developing with the SUID sandbox. If you want to live dangerously and need an immediate workaround, you can try using --no-sandbox.
#0 0x5638b320a259 base::debug::CollectStackTrace()
#1 0x5638b31769e3 base::debug::StackTrace::StackTrace()
#2 0x5638b3187610 logging::LogMessage::~LogMessage()
#3 0x5638b187083b content::ZygoteHostImpl::Init()
#4 0x5638b311680c content::ContentMainRunnerImpl::Initialize()
#5 0x5638b3114b09 content::RunContentProcess()
#6 0x5638b3114c3d content::ContentMain()
#7 0x5638b3170c29 headless::HeadlessBrowserMain()
#8 0x5638b317093a headless::HeadlessShellMain()
#9 0x5638b01a13e1 ChromeMain
#10 0x7f436fc2a1ca (/usr/lib/x86_64-linux-gnu/libc.so.6+0x2a1c9)
#11 0x7f436fc2a28b __libc_start_main
#[12](https://github.com/NCIOCPL/ncids/actions/runs/13640631059/job/38129490135#step:16:13) 0x5638b01a122a _start

Received signal 6
#0 0x5638b320a259 base::debug::CollectStackTrace()
#1 0x5638b31769e3 base::debug::StackTrace::StackTrace()
#2 0x5638b3209e00 base::debug::(anonymous namespace)::StackDumpSignalHandler()
#3 0x7f436fc45330 (/usr/lib/x86_64-linux-gnu/libc.so.6+0x4532f)
#4 0x7f436fc9eb2c pthread_kill
#5 0x7f436fc4527e gsignal
#6 0x7f436fc288ff abort
#7 0x5638b3209425 base::debug::BreakDebugger()
#8 0x5638b3187a37 logging::LogMessage::~LogMessage()
#9 0x5638b187083b content::ZygoteHostImpl::Init()
#10 0x5638b311680c content::ContentMainRunnerImpl::Initialize()
#11 0x5638b3114b09 content::RunContentProcess()
#12 0x5638b3114c3d content::ContentMain()
#[13](https://github.com/NCIOCPL/ncids/actions/runs/13640631059/job/38129490135#step:16:14) 0x5638b3170c29 headless::HeadlessBrowserMain()
#[14](https://github.com/NCIOCPL/ncids/actions/runs/13640631059/job/38129490135#step:16:15) 0x5638b317093a headless::HeadlessShellMain()
#[15](https://github.com/NCIOCPL/ncids/actions/runs/13640631059/job/38129490135#step:16:16) 0x5638b01a13e1 ChromeMain
#[16](https://github.com/NCIOCPL/ncids/actions/runs/13640631059/job/38129490135#step:16:17) 0x7f436fc2a1ca (/usr/lib/x86_64-linux-gnu/libc.so.6+0x2a1c9)
#[17](https://github.com/NCIOCPL/ncids/actions/runs/13640631059/job/38129490135#step:16:18) 0x7f436fc2a28b __libc_start_main
#[18](https://github.com/NCIOCPL/ncids/actions/runs/13640631059/job/38129490135#step:16:19) 0x5638b01a122a _start
  r8: 0000000000000000  r9: 0000000000000000 r10: 0000000000000008 r11: 0000000000000246
 r12: 0000000000000006 r13: 00007ffd7cd7c2f0 r14: 0000000000000016 r15: aaaaaaaaaaaaaaaa
  di: 000000000000153d  si: 000000000000153d  bp: 00007ffd7cd7c1f0  bx: 000000000000153d
  dx: 0000000000000006  ax: 0000000000000000  cx: 00007f436fc9eb2c  sp: 00007ffd7cd7c1b0
  ip: 00007f436fc9eb2c efl: 0000000000000246 cgf: 002b000000000033 erf: 0000000000000000
 trp: 0000000000000000 msk: 0000000000000000 cr2: 0000000000000000
[end of stack trace]
Calling _exit(1). Core file will not be generated.


TROUBLESHOOTING: https://github.com/puppeteer/puppeteer/blob/main/docs/troubleshooting.md

    at onClose (/home/runner/work/ncids/ncids/node_modules/.pnpm/puppeteer@9.1.1_encoding@0.1.13/node_modules/puppeteer/lib/cjs/puppeteer/node/BrowserRunner.js:[19](https://github.com/NCIOCPL/ncids/actions/runs/13640631059/job/38129490135#step:16:20)4:20)
    at Interface.<anonymous> (/home/runner/work/ncids/ncids/node_modules/.pnpm/puppeteer@9.1.1_encoding@0.1.13/node_modules/puppeteer/lib/cjs/puppeteer/node/BrowserRunner.js:184:68)
    at Interface.emit (node:events:530:35)
    at Interface.close (node:internal/readline/interface:527:10)
    at Socket.onend (node:internal/readline/interface:253:10)
    at Socket.emit (node:events:530:35)
    at endReadableNT (node:internal/streams/readable:1698:12)
    at process.processTicksAndRejections (node:internal/process/task_queues:82:21)

Node.js v[20](https://github.com/NCIOCPL/ncids/actions/runs/13640631059/job/38129490135#step:16:21).18.3
 ELIFECYCLE  Command failed with exit code 1.
Error: Process completed with exit code 1.
```

--- a/.github/workflows/workflow.yml
+++ b/.github/workflows/workflow.yml
@@ -15,7 +15,7 @@ on:
 jobs:
   build:
     name: Build, Test and Upload Artifacts
-    runs-on: ubuntu-20.04
+    runs-on: ubuntu-22.04
     services:
       apache:
         image: httpd:2.4

--- a/testing/ncids-js-testing/.pa11yci.js
+++ b/testing/ncids-js-testing/.pa11yci.js
@@ -1,6 +1,7 @@
 const glob = require('glob');
 const path = require('path');
 const paths = require('./config/paths');
+const { exec } = require('child_process');
 
 const htmlFiles = glob.sync(
 	path.join(paths.public, '**', '*.html'),
@@ -27,16 +28,25 @@ const urls = htmlFiles.map(
 	},
 );
 
+// Starting with Ubuntu 22.04, pa11y-ci requires us to tell the Chrome binary's path
+// Fortunately, GitHub Actions has an environment variable for that.
+// The rest of this is so we don't break local development on Macs.
+// According to the (current) docs, this can all go away once we upgrade pa11y-ci to 4.0 .
+const defaultChromeLaunchConfig = {
+	args: [
+		'--no-sandbox',
+		'--disable-setuid-sandbox',
+		'--disable-dev-shm-usage',
+	],
+};
+const chromeLaunchConfig = process.env.CHROME_BIN
+	? { executablePath: process.env.CHROME_BIN, ...defaultChromeLaunchConfig }
+	: defaultChromeLaunchConfig;
+
 module.exports = {
 	defaults: {
 		standard: 'WCAG2AA', // Default is WCAG2AA, but leaving this here as a note.
 	},
 	urls: urls,
-	chromeLaunchConfig: {
-		args: [
-			'--no-sandbox',
-			'--disable-setuid-sandbox',
-			'--disable-dev-shm-usage',
-		],
-	},
+	chromeLaunchConfig: chromeLaunchConfig,
 };

